airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py

```diff
@@ -107,12 +112,45 @@ def get_incremental_stream_config(google_api: GoogleAds, config: Mapping[str, An
         )
         return incremental_stream_config
 
-    @staticmethod
-    def get_account_info(google_api: GoogleAds, config: Mapping[str, Any]) -> Iterable[Iterable[Mapping[str, Any]]]:
-        dummy_customers = [CustomerModel(id=_id) for _id in config["customer_id"].split(",")]
-        accounts_stream = ServiceAccounts(google_api, customers=dummy_customers)
-        for slice_ in accounts_stream.stream_slices():
-            yield accounts_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=slice_)
+    def get_all_accounts(self, google_api: GoogleAds, customers: List[CustomerModel], customer_status_filter: List[str]) -> List[str]:
+        customer_clients_stream = CustomerClient(api=google_api, customers=customers, customer_status_filter=customer_status_filter)
+        for slice in customer_clients_stream.stream_slices():
+            for record in customer_clients_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=slice):
+                yield record
+
+    def _get_all_connected_accounts(
+        self, google_api: GoogleAds, customer_status_filter: List[str]
+    ) -> Iterable[Iterable[Mapping[str, Any]]]:
+        customer_ids = [customer_id for customer_id in google_api.get_accessible_accounts()]
+        dummy_customers = [CustomerModel(id=_id, login_customer_id=_id) for _id in customer_ids]
+
+        yield from self.get_all_accounts(google_api, dummy_customers, customer_status_filter)
+
+    def get_customers(self, google_api: GoogleAds, config: Mapping[str, Any]) -> List[CustomerModel]:
+        customer_status_filter = config.get("customer_status_filter", [])
+        accounts = self._get_all_connected_accounts(google_api, customer_status_filter)
+        customers = CustomerModel.from_accounts(accounts)
+
+        # filter duplicates as one customer can be accessible from mutiple connected accounts
+        unique_customers = []
+        seen_ids = set()
+        for customer in customers:
+            if customer.id in seen_ids:
+                continue
+            seen_ids.add(customer.id)
+            unique_customers.append(customer)
+        customers = unique_customers
+        customers_dict = {customer.id: customer for customer in customers}
+
+        # filter only selected accounts
+        if config.get("customer_ids"):
+            customers = []
+            for customer_id in config["customer_ids"]:
+                if customer_id not in customers_dict:
+                    logging.warning(f"Customer with id {customer_id} is not accessible. Skipping it.")
```

Comment:
The previous version raised an error when accessing data for inaccessible customer IDs. This was because the connector could not retrieve data from disabled accounts. In the new version, I've updated the behaviour to enhance robustness. Now, instead of failing, the connector logs a warning when it encounters inaccessible accounts. This change aims to avoid errors due to disabled accounts.